addon/components/stacked-vertical-bar-chart.js

```diff
@@ -0,0 +1,1015 @@
+    if (!this.get('barSortAscending')) {
+      sortedBarNames.reverse();
+    }
+    return sortedBarNames;
+  }),
+
+  /**
+   * Explicitly written version of the default comparison function that is used
+   * by Array#sort. Used by every slice and bar comparison functions that are
+   * comparing specific parameters.
+   * @function
+   */
+  defaultCompareFn: function(reference1, reference2) {
+    if (reference1 < reference2) {
+      return -1;
+    } else if (reference1 > reference2) {
+      return 1;
+    } else {
+      return 0;
+    }
+  },
+
+  // ---------------------------------------------------------------------------
+  // Layout
+  // ---------------------------------------------------------------------------
+
+  labelHeightOffset: Ember.computed('_shouldRotateLabels', 'maxLabelHeight', 'labelHeight', 'labelPadding', function() {
+    var labelSize;
+
+    if (this.get('_shouldRotateLabels')) {
+      labelSize = this.get('maxLabelHeight');
+    } else {
+      labelSize = this.get('labelHeight');
+    }
+    return labelSize + this.get('labelPadding');
+  }),
+
+  // Chart Graphic Dimensions
+  graphicLeft: Ember.computed.alias('labelWidthOffset'),
+
+  graphicWidth: Ember.computed('width', 'labelWidthOffset', function() {
+     return this.get('width') - this.get('labelWidthOffset');
+  }),
+
+  graphicHeight: Ember.computed('height', 'legendHeight', 'legendChartPadding', function() {
+    return this.get('height') - this.get('legendHeight') -
+      this.get('legendChartPadding');
+  }),
+
+  // ---------------------------------------------------------------------------
+  // Ticks and Scales
+  // ---------------------------------------------------------------------------
+
+  // Vertical position/length of each bar and its value
+  yDomain: Ember.computed('finishedData', function() {
+    var finishedData = this.get('finishedData');
+
+    var max = d3.max(finishedData, function(d) {
+      return d.max;
+    });
+
+    var min = d3.min(finishedData, function(d) {
+      return d.min;
+    });
+
+    // force one end of the range to include zero
+    if (min > 0) {
+      return [0, max];
+    }
+    if (max < 0) {
+      return [min, 0];
+    }
+    if (min === 0 && max === 0) {
+      return [0, 1];
+    } else {
+      return [min, max];
+    }
+  }),
+
+  yScale: Ember.computed('graphicTop', 'graphicHeight', 'yDomain', 'numYTicks', function() {
+    return d3.scale.linear()
+      .domain(this.get('yDomain'))
+      .range([ this.get('graphicTop') + this.get('graphicHeight'),
+               this.get('graphicTop') ])
+      .nice(this.get('numYTicks'));
+  }),
+
+  /**
+   * All slice labels to show in the chart legend. Includes 'Other' slice if
+   * the 'Other' slice is present.
+   * @type {Array.<string>}
+   */
+  allSliceLabels: Ember.computed('nonOtherSliceTypes.[]', 'otherSliceTypes.[]', 'otherSliceLabel', function() {
+    var result = _.clone(this.get('nonOtherSliceTypes'));
```

Comment:
Why do you need to clone here? Is this a deep clone? I think you need to start with empty array and then push all `nonOtherSliceTypes` objects into that array.